bittensor/utils/__init__.py

```diff
@@ -95,42 +98,43 @@ def solve_for_difficulty( block_hash, difficulty ):
             break
     return nonce, seal
 
-
-def solve_for_difficulty_fast( subtensor ):
+def solve_for_difficulty_fast( subtensor, num_processes: int = 5, update_interval: int = 100000 ) -> None:
+    """
+    Solves the POW for registration using multiprocessing.
+    Args:
+        subtensor
+            Subtensor to connect to for block information and to submit.
+        num_processes: int
+            Number of processes to use.
+        update_interval: int
+            Number of nonces to solve before updating block information.
+    Note: 
+    - We should modify the number of processes based on user input.
+    - We can also modify the update interval to do smaller blocks of work,
+        while still updating the block information after a different number of nonces,
+        to increase the transparency of the process while still keeping the speed.
+    """
     block_number = subtensor.get_current_block()
     difficulty = subtensor.difficulty
     block_hash = subtensor.substrate.get_block_hash( block_number )
     while block_hash == None:
         block_hash = subtensor.substrate.get_block_hash( block_number )
     block_bytes = block_hash.encode('utf-8')[2:]
     
-    meets = False
     nonce = -1
     limit = int(math.pow(2,256) - 1)
-    best = math.inf
-    update_interval = 100000
     start_time = time.time()
 
     console = bittensor.__console__
-    with console.status("Solving") as status:
-        while not meets:
-            nonce += 1 
-
-            # Create seal.
-            nonce_bytes = binascii.hexlify(nonce.to_bytes(8, 'little'))
-            pre_seal = nonce_bytes + block_bytes
-            seal = hashlib.sha256( bytearray(hex_bytes_to_u8_list(pre_seal)) ).digest()
-
-            seal_number = int.from_bytes(seal, "big")
-            product = seal_number * difficulty
-            if product - limit < best:
-                best = product - limit
-                best_seal = seal
-
-            if product < limit:
-                return nonce, block_number, block_hash, difficulty, seal
-
-            if nonce % update_interval == 0:
+    with console.status("Solving") as status:        
+        found_solution = multiprocessing.Value('q', -1, lock=False) # int
```

Comment:
I think the subtensor connection is not multiprocess safe. Have you checked this.

Reply:
I have not.
However,
The multiprocessing stops every update_interval as the block of work is completed. The loop then checks if a solution was found (any solution) and submits the solution.
Therefore, this doesn't access the subtensor while multiprocessed.

Reply:
Ok, that's good. However you can also just create a new connection in each process if you need to.